### PR TITLE
Add Needed Pages

### PR DESCRIPTION
**Pull request by lqthai07t3**
- Define needed pages.
- Create pages.
- Fix some conflict after creating pages.
- Run code and check.

image

### CODE REVIEW

File: TADashboard/src/main/java/base_pages/BasePage.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    public boolean doesWelcomeLabelDisplay() {
+        return lblWelcome.isVisible();
+    }
+
+    public boolean doesAdministerLabelDisplay() {
+        return lblAdminister.isVisible();
+    }
+
+    public String getAlertText() {
```
Comment (MinhQuan119) [inline TADashboard/src/main/java/base_pages/BasePage.java:32]: don't need to create method for it, same for acceptAlert

Comment (tanmle) [inline TADashboard/src/main/java/base_pages/BasePage.java:32]: again, I mentioned about this many times, we don't need to wrap a method for the DriverUtils method.

File: TADashboard/src/main/java/base_pages/GeneralSettings_BasePage.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package base_pages;
+
+import com.logigear.control.common.imp.Button;
+import com.logigear.control.common.imp.Label;
+
+public class GeneralSettings_BasePage extends BasePage {
```
Comment (tanmle) [inline TADashboard/src/main/java/base_pages/GeneralSettings_BasePage.java:6]: what is the difference between GeneralSetting and BasePage? and what naming convention of a class is this?

File: TADashboard/src/main/java/base_pages/GeneralSettings_BasePage.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    private Button btnFinish = new Button("css=input[value='Finish']");
+    private Button btnCancel = new Button("css=input[value='Cancel']");
+    private Label dynamicLabel = new Label("xpath=//ul[@id='wstep']/li[text()='%s']");
```
Comment (tanmle) [inline TADashboard/src/main/java/base_pages/GeneralSettings_BasePage.java:9]: can you show me where I can find this dynamic label in the UI?

File: TADashboard/src/main/java/page_objects/LoginPage.java [hunk 4]
```diff
@@ -1,9 +1,10 @@
 package page_objects;
 
+import base_pages.BasePage;
 import com.logigear.control.common.imp.Button;
 import com.logigear.control.common.imp.TextBox;
 
-public class LoginPage {
+public class LoginPage extends BasePage {
```
Comment (MinhQuan119) [inline TADashboard/src/main/java/page_objects/LoginPage.java:7]: why does LoginPage extends BasePage?

File: TADashboard/src/main/java/page_objects/DataProfilesPage.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import utils.common.Constants;
+
+public class DataProfilesPage extends BasePage {
+    private CheckBox dynamicCheckbox = new CheckBox("xpath=//table[@class='GridView']/tbody/tr/td[text()='%s']/preceding-sibling::td/input");
+    private Link dynamicDataProfile = new Link("xpath=//table[@class='GridView']/tbody/tr/td[text()='%s']");
+    private Link dynamicAction = new Link("xpath=//table[@class='GridView']/tbody/tr/td[text()='%s']/following::td[@class='center']/a/parent::td");
+    private Link lnkAddNew = new Link("css=div[class='panel_tag2'] a[href='profile.jsp?action=create']");
+    private Link lnkDataProfiles = new Link("xpath=//table[@class='GridView']/tbody/tr/td[count(//tbody/tr/th) - count(//tbody/tr/th[text()='Data Profile']/following-sibling::th)]");
+
+    public String getTextOfAction(String string) {
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/DataProfilesPage.java:15]: what does `string` mean?

File: TADashboard/src/main/java/base_pages/GeneralSettings_BasePage.java (hunk 2 above)
Comment (tanmle) [reply TADashboard/src/main/java/base_pages/GeneralSettings_BasePage.java:6]: this naming is not acceptable

**Review by tanmle — CHANGES_REQUESTED** [review]

**Pull request closed without merging**